src/main/java/org/folio/circulation/domain/AccountRepository.java

```diff
@@ -43,16 +43,17 @@ public AccountRepository(Clients clients) {
     feefineActionsStorageClient = clients.feeFineActionsStorageClient();
   }
 
-  public CompletableFuture<Result<Loan>> findAccountsForLoan(Result<Loan> loanResult) {
+  public CompletableFuture<Result<Loan>> findAccountsAndActionsForLoan(Result<Loan> loanResult) {
```

Comment:
Are actions considered separate to accounts? If they are not, why include them in the name of this method?

Reply:
The `action` part was added in order to distinguish method that fetches accounts only vs accounts + fee/fine actions for them.

Reply:
Ah, that classic challenge, that's been causing pain in mod-circulation for some time.

In what circumstances are accounts needed without actions?

Reply:
In the case currently being implemented. We already have all the needed information in account itself. 

Hmm... I think in most of the cases we don't need action history. The action history is required for refunds, i think.

Reply:
Ok. Maybe we need to represent this distinction with types as well as method names, so that it is less ambiguous whether a client needs that information or not.

That may well be out of scope of this change.

Reply:
I've raised https://issues.folio.org/browse/CIRC-754 for this.

Reply:
Thank you